File: go/memory/memory_benchmark.go
package main

import (
	"fmt"
	"time"
)

func main() {
	start := time.Now()
	N := 10_000_000
	arr := make([]int, N)
	for i := 0; i < N; i++ {
		arr[i] = i * 2
	}
	sum := 0
	for _, v := range arr {
		sum += v
	}
	fmt.Printf("Memory Benchmark: Sum: %d | Time: %v\n", sum, time.Since(start))
}

File: node/memory-benchmark.ts
const s = Date.now();
const n = 10_000_000;
const arr = new Array(n).fill(0).map((_, i) => i * 2);
const sum = arr.reduce((acc, v) => acc + v, 0);
console.log(`Memory Benchmark: Sum: ${sum} | Time: ${Date.now() - s}ms`);
